lib/writer/block-write-stream.js

```diff
@@ -95,6 +97,27 @@ class BlockWriteStream extends stream.Writable {
     this.open()
   }
 
+  /**
+   * @summary Determine whether an error is considered a
+   * transient occurrence, and the operation should be retried
+   * Errors considered potentially temporary are:
+   *   - Mac OS: ENXIO, EBUSY
+   *   - Windows: ENOENT, UNKNOWN
+   *   - Linux: EIO
```

Comment:
So this comment implies that each OS has unique error codes... Is that correct, or could Linux throw a `ENOENT` in a different situation, and we'd want to treat that as a non-transient error? (i.e. do we need to check both the error code **and** the current OS?)

Reply:
Correct. We should check for the platform as in https://github.com/resin-io-modules/etcher-image-write/blob/master/lib/filesystem.js#L75

Reply:
Done